clap_derive/Cargo.toml

```diff
@@ -42,6 +42,7 @@ quote = "1.0.9"
 proc-macro2 = "1.0.42"
 heck = "0.4.0"
 proc-macro-error = "1"
+pulldown-cmark = "0.9"
```

Comment:
> minimad doesn't parse blocks and only parses some inlines. To be able to use this we'd have the write the first half of a markdown parser anyway.

I wonder how much of a problem that will actually be, especially compared to what we do now.  The build times are significantly different than pulldown

https://github.com/rosetta-rs/md-rosetta-rs

Reply:
Yep, I get that. Unfortunately, the way that it accomplishes that is by not really being a markdown parser. It's a parser for something that sorta looks like a subset of markdown.

Honestly I think that not supporting markdown is a better option than supporting `minimad`'s version of "markdown".

---

There's some things that look like they're just bugs that we could fix that I can only assume would be accepted, like:

Inlines aren't parsed in lists:

```
* some bullet item
* some _bullet_ item
* some bullet item
```

```
Normal(Composite { style: ListItem, compounds: ["some bullet item"] })
Normal(Composite { style: ListItem, compounds: ["some _bullet_ item"] })
Normal(Composite { style: ListItem, compounds: ["some bullet item"] })
```

There's some things that I'm not sure if they would consider them to be a bug, but we could technically workaround:

Each line of a paragraph is considered a paragraph.

```
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
eiusmod tempor incididunt ut labore et dolore magna aliqua. Dictum at
tempor commodo ullamcorper a lacus vestibulum sed arcu.  
```
```
Normal(Composite { style: Paragraph, compounds: ["Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do"] })
Normal(Composite { style: Paragraph, compounds: ["eiusmod tempor incididunt ut labore et dolore magna aliqua. Dictum at"] })
Normal(Composite { style: Paragraph, compounds: ["tempor commodo ullamcorper a lacus vestibulum sed arcu."] })
```

Sub-lists (bonus points for parsing a single unconnected `*` as the start of an italic block)

```
* some bullet item
  * some bullet item
* some bullet item
```
```
Normal(Composite { style: ListItem, compounds: ["some bullet item"] })
Normal(Composite { style: Paragraph, compounds: ["  ", I" some bullet item"] })
Normal(Composite { style: ListItem, compounds: ["some bullet item"] })
```

Only one block level is supported:

```
> # foobar
```

```
Normal(Composite { style: Quote, compounds: ["# foobar"] })
```

There's way more, these were just the most obviously wrong of the few things I tried. Honestly I think it'd be easier and faster to write a new markdown parser than to pre-parse and re-parse around this crate to get reasonable output of any sort of basic markdown _document_. That would put a major dent in its lead in your stats too.

---

**tl;dr**: We should heed the warning on that crate that says:

> If you're looking for a Markdown parser, this one is probably *not* the one you want

What you want to use it for isn't what it's meant to be used for. It's not a general purpose markdown parser. IMO, it's not even a "general purpose (when your purpose is outputting to the terminal) markdown parser". It's good for when you want some minimal style in terminal output of static, non-user-supplied, text.